### PR TITLE
Noam/dee 511 write check docs traintestperformance

### PR DESCRIPTION
**Pull request by noamzbr**
Add plot example to train test performance in nlp

### CODE REVIEW

File: docs/source/checks/nlp/model_evaluation/plot_train_test_performance.py [hunk 1]
```diff
@@ -0,0 +1,81 @@
+# -*- coding: utf-8 -*-
+"""
+.. _plot_nlp_train_test_performance:
```
Comment (Nadav-Barak) [inline docs/source/checks/nlp/model_evaluation/plot_train_test_performance.py:3]:
```suggestion
.. _nlp__train_test_performance:
```

File: docs/source/checks/nlp/model_evaluation/plot_train_test_performance.py [hunk 2]
```diff
@@ -0,0 +1,81 @@
+# -*- coding: utf-8 -*-
+"""
+.. _nlp__train_test_performance:
+
+Train Test Performance for NLP Models
+**************************************
+This notebook provides an overview for using and understanding train test performance check.
```
Comment (shir22) [inline docs/source/checks/nlp/model_evaluation/plot_train_test_performance.py:7]:
```suggestion
This notebook provides an overview for using and understanding the train test performance check.
```

File: docs/source/checks/nlp/model_evaluation/plot_train_test_performance.py [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+What is the purpose of the check?
+==================================
+This check helps you compare your NLP model's performance between the train and test datasets based on multiple metrics.
+
+For Text Classification tasks the supported metrics are sklearn scorers. You may use any of the existing sklearn
+scorers or create your own. For more information about the supported sklearn scorers, defining your own metrics and
+to learn how to use metrics for other supported task types, see the
+:doc:`Metrics Guide </user-guide/general/metrics_guide>`.
+
+The default scorers that are used for are F1, Precision, and Recall for Classification,
```
Comment (shir22) [inline docs/source/checks/nlp/model_evaluation/plot_train_test_performance.py:26]: delete the "for" / add a word like "evaluation" after it

File: docs/source/checks/nlp/model_evaluation/plot_train_test_performance.py [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+and F1 Macro, Recall Macro and Precision Macro for Token Classification. See more about the supported task types at the
+:doc:`Supported Tasks </user-guide/nlp/supported_tasks>` guide.
```
Comment (shir22) [inline docs/source/checks/nlp/model_evaluation/plot_train_test_performance.py:28]: @Nadav-Barak , @noamzbr  - Ensure this PR is merged after the docs restructure... (?)
(so I guess also the link for supported tasks should be updated...)

File: docs/source/checks/nlp/model_evaluation/plot_train_test_performance.py [hunk 5]
```diff
@@ -0,0 +1,81 @@
...
+
+train_dataset, test_dataset = load_data()
+train_preds, test_preds = load_precalculated_predictions('predictions')
+
+#%%
+# Run the check
+# ==============
+#
+# You can select which scorers to use by passing either a list or a dict of scorers to the check,
+# the full list of possible scorers can be seen at scorers.py.
```
Comment (shir22) [inline docs/source/checks/nlp/model_evaluation/plot_train_test_performance.py:46]: can we add a link to the api reference if one exists for the scorers? (or in general - don't we something that we alway use to conveniently link to scorers etc.? - like link also to the the metrics guide / copy-paste a relevant snippet from there?)

File: docs/source/checks/nlp/model_evaluation/plot_single_dataset_performance.py [hunk 6]
```diff
@@ -17,12 +17,14 @@
...
-
-The default scorers that are used are F1, Precision, and Recall.
+For Text Classification tasks the supported metrics are sklearn scorers. You may use any of the existing sklearn
+scorers or create your own. For more information about the supported sklearn scorers, defining your own metrics and
+to learn how to use metrics for other supported task types, see the
+:doc:`Metrics Guide </user-guide/general/metrics_guide>`.
+
+The default scorers that are used for are F1, Precision, and Recall for Classification,
+and F1 Macro, Recall Macro and Precision Macro for Token Classification. See more about the supported task types at the
+:doc:`Supported Tasks </user-guide/nlp/supported_tasks>` guide.
```
Comment (shir22) [inline docs/source/checks/nlp/model_evaluation/plot_single_dataset_performance.py:27]: same comment as following on links and prs order etc.

File: docs/source/checks/nlp/model_evaluation/plot_train_test_performance.py (hunk 5 above)
Comment (noamzbr) [reply docs/source/checks/nlp/model_evaluation/plot_train_test_performance.py:46]: I think linking to the metric guide is the best option.

File: docs/source/checks/nlp/train_test_validation/plot_property_drift.py [hunk 7]
```diff
@@ -0,0 +1,102 @@
...
+Which NLP Properties Are Used?
+-------------------------------
+
+The properties used by this check are the ones present in datasets prior to running this check. That's why it's
+important to first run ``calculate_default_properties`` on both the train and test datasets prior to running this
+check. Note also that in order to calculate drift on more complex properties, such as text Toxicity or Fluency,
+you need to pass the ``include_long_calculation_properties=True`` parameter to ``calculate_default_properties``. For
+more information on how to calculate properties, please visit the :doc:`nlp properties guide
+</user-guide/nlp/nlp_properties>`.
+
```
Comment (nirhutnik) [inline docs/source/checks/nlp/train_test_validation/plot_property_drift.py:45]: If you're missing some "content" at the beginning of the file, I would suggest adding some explanations on WHY we're looking for drift in the properties (not being able to check on the text itself).

File: docs/source/checks/nlp/train_test_validation/plot_property_drift.py [hunk 8]
```diff
@@ -0,0 +1,102 @@
...
+=============
+"""
+
+from deepchecks.nlp.datasets.classification.tweet_emotion import load_data
+
+train_dataset, test_dataset = load_data()
+
+# # Calculate properties, commented out because it takes a short while to run
+# train_dataset.calculate_default_properties(include_long_calculation_properties=True)
+# test_dataset.calculate_default_properties(include_long_calculation_properties=True)
```
Comment (nirhutnik) [inline docs/source/checks/nlp/train_test_validation/plot_property_drift.py:56]: Nice, not sure that it's needed but not critical to me

File: docs/source/checks/nlp/train_test_validation/plot_property_drift.py [hunk 9]
```diff
@@ -0,0 +1,102 @@
...
+#  check_result.show_in_window()
+#%%
+# The result will be displayed in a new window.
+
+#%%
+# Observe the check’s output
+# --------------------------
+# The result value is a dict that contains drift score and method used for each text property.
+
+check_result.value
```
Comment (nirhutnik) [inline docs/source/checks/nlp/train_test_validation/plot_property_drift.py:78]: Maybe add a "what we can see and what it means" cell here.
I guess there's natural drift here?

File: docs/source/checks/nlp/train_test_validation/plot_property_drift.py (hunk 7 above)
Comment (noamzbr) [reply docs/source/checks/nlp/train_test_validation/plot_property_drift.py:45]: added something

File: docs/source/checks/nlp/train_test_validation/plot_property_drift.py (hunk 8 above)
Comment (noamzbr) [reply docs/source/checks/nlp/train_test_validation/plot_property_drift.py:56]: Should be in the property docs IMO
